### PR TITLE
Only send avoidance heartbeat when Safe Landing is fully enabled

### PR DESCRIPTION
**Pull request by DanMesh**
#### Problem

When the Autopilot Manager is running, it sent out a heartbeat for the Obstacle Avoidance (OA) system (component 196) even when OA had been disabled or Safe Landing (SL) was inactive.
This caused the OA indicator in the AMC top bar to show, even though the OA ought to be disabled and "out of sight".

#### Changes

This sets the heartbeat to only be sent when OA is enabled and SL is active.
This meant that some of the logic about when to check whether to start a landing site search could be simplified.

Furthermore, the heartbeat is now only sent when all important components are healthy, specifically:
- the Sensor Manager is receiving image and odometry data,
- the Landing Manager is receiving images and odometry from the Sensor Manager, and
- the Misison Manager is receiving a stream of OA interface wayppoint messages.

#### Testing
SITL

_OA indicator:_
- The OA indicator is only visible when OA is enabled (`COM_OBS_AVOID == 1`).
   - Green when SL is enabled and running.
   - Red when SL is disabled or not running.
- There seems to be an issue in AMC:
   - When OA is switched from disabled to enabled,the indicator does not appear (incorrect).
   - When OA is switched from enabled to disabled,the indicator disappears (correct).

_Enabling/disabling logic:_
- Landing site search only kicks in if OA and SL are enabled.
- Disabling OA while a landing site search is being executed results in a hold.
- Take-off is denied if OA is enabled but SL is either disabled or not running correctly.

---

**Jira Task:** [AVOID-292](https://auterion.atlassian.net/browse/AVOID-292) & [AVOID-293](https://auterion.atlassian.net/browse/AVOID-293)

**Does this PR need to be backported to the stable release?** No

**Does this PR need to update any documentation (readme, confluence, gitbook, etc.)?** No

[AVOID-292]: https://auterion.atlassian.net/browse/AVOID-292?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/AutopilotManager.cpp [hunk 1]
```diff
@@ -411,9 +411,15 @@ void AutopilotManager::start() {
         _mission_manager->init();
         _mission_manager_th = std::thread(&AutopilotManager::run_mission_manager, this);
 
+        // MAVLink passthrough
```
Comment (frederictaillandier) [inline src/AutopilotManager.cpp:414]: It s starting to be a very big function.
It' s just a suggestion but maybe you would need to subdivide it into smaller pieces or use an object to build all that stuff ?

File: src/modules/sensor_manager/SensorManager.cpp [hunk 2]
```diff
@@ -149,7 +150,8 @@ auto SensorManager::run() -> void {
         _time_sync.run(tsync.ts1, tsync.tc1, this->now().nanoseconds() / 1000ULL);
     });
 
-    _timer_status_task = create_wall_timer(100ms, std::bind(&SensorManager::health_check, this));
+    _timer_health_check_task = create_wall_timer(100ms, std::bind(&SensorManager::health_check, this));
```
Comment (frederictaillandier) [inline src/modules/sensor_manager/SensorManager.cpp:153]: DO you want to save these 100ms in a constexpr on top of the file you could modify later ?

**Review by frederictaillandier — APPROVED** [review]
LGTM

File: src/AutopilotManager.cpp (hunk 1 above)
Comment (DanMesh) [reply src/AutopilotManager.cpp:414]: You're right, it's getting quite hard to follow.
I'll tidy it up and put together a follow-up PR to keep this one from getting too big.
